community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/LockManagerImpl.java

```diff
@@ -168,11 +168,14 @@ private RWLock getRWLockForReleasing( Object resource, Object tx, int readCountP
                 throw new LockNotFoundException( "Lock not found for: "
                     + resource + " tx:" + tx );
             }
-            if ( !lock.isMarked() && lock.getReadCount() == readCountPrerequisite &&
-                lock.getWriteCount() == writeCountPrerequisite &&
-                lock.getWaitingThreadsCount() == 0 )
+            synchronized ( lock )
```

Comment:
A comment here saying which other access to lock this is expected to synchonize with would be useful.